app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt

```diff
@@ -9,13 +8,25 @@ import android.widget.ImageView
 import android.widget.LinearLayout
 import com.nilhcem.blenamebadge.R
 import com.nilhcem.blenamebadge.data.DrawableInfo
+import com.nilhcem.blenamebadge.util.Resource
+
+class DrawableAdapter : RecyclerView.Adapter<DrawableAdapter.DrawableItemHolder>() {
+    companion object {
+        private var INSTANCE: DrawableAdapter? = null
+
+        fun getInstance(): DrawableAdapter {
+            if (INSTANCE == null) INSTANCE = DrawableAdapter()
+            return INSTANCE as DrawableAdapter
+        }
+    }
 
-class DrawableAdapter(private val context: Context?, private val list: List<DrawableInfo>) : RecyclerView.Adapter<DrawableAdapter.DrawableItemHolder>() {
     private var selectedPosition: Int = -1
     private var listener: OnDrawableSelected? = null
+    private val drawableList = ArrayList<DrawableInfo>()
```

Comment:
Lets stick to List<DrawableInfo>

Reply:
List is an abstract, I think it can't be initialized..

Reply:
```suggestion
    private val drawableList : List<DrawableInfo> = mutableListOf()
```

Reply:
Why should we initialize a MutableList instead of ArrayList? Because I don't find any problems with ArrayList for now..

Reply:
@iamareebjamal suggested me to use List instead of arraylist, as we are not using any feature of arraylist, then why use it in the first place. Lets wait for his reply on this

Reply:
mutableList is kotlin specific. ArrayList is java. You can use mutableList in Kotlin/Native iOS, not ArrayList